src/org/opendatakit/briefcase/export/ExportConfiguration.java

```diff
@@ -144,17 +163,41 @@ public ExportConfiguration setExportDir(Path path) {
     return this;
   }
 
+  public ExportConfiguration removeExportDir() {
+    this.exportDir = Optional.empty();
+    return this;
+  }
+
   public Optional<Path> getPemFile() {
     return pemFile;
   }
 
   public ExportConfiguration setPemFile(Path path) {
-    this.pemFile = Optional.ofNullable(path);
+    //Check if path is a file and is readable
+    if (Files.isRegularFile(path) && Files.isReadable(path)) {
+      //todo check if file is parsable by PEMParser and contains private key
+      this.pemFile = Optional.ofNullable(path);
+    }
```

Comment:
What if we don't want to set it? How does the UI know that this has happened?

Reply:
Still working on this. This logic will be ported in the PrivateKey class which will contain a static method to check the validity of a pem file(path). This method will be used at the UI level to verify if the path provided is a valid pem if not an error message will be shown and the path will be rejected.

Reply:
Good point @icemc! I'd suggest another name for the class because `PrivateKey` already exists and we're already using it on the export processes.

I think it's a great idea to use the new validation you're going to define as soon as possible. You could define a new filter for the FileChooser instance if you want (maybe you already thought of this).

If you do that, I'd change the member `ExportConfiguration.pemFile` for a `Optional<PrivateKey> privateKey` member. Otherwise, if you maintain the `pemFile` member, you'll have to validate it again in the setter. Every instance of `ExportConfiguration` must be always valid. 

If the UI already validates the pemFile location, it can extract the `PrivateKey` instance (maybe on the new class you're going to extract) and pass it to the `ExportConfiguration` guaranteeing that's always valid.

Also, remember to write unit tests for the new class that will handle cryptographic stuff.